source/dubregistry/web.d

```diff
@@ -139,6 +140,42 @@ class DubRegistryWebFrontend {
 		getPackageVersion(req, res, _packname, null);
 	}
 
+	@path("/packages/:packname/logo")
+	void getPackageLogo(HTTPServerRequest req, HTTPServerResponse res, string _packname)
+	{
+		string rev = req.queryString;
+		if (!rev.length)
+		{
+			// get latest revision of file
+			auto logo = m_registry.getPackageLogo(_packname);
+			if (logo == req.requestPath.toString)
+				throw new Exception("Invalid package logo");
+			if (logo.length)
+			{
+				res.redirect(logo);
+				return;
+			}
+		}
+
+		string logo = _packname ~ "@" ~ rev;
+
+		bool[logoFormats.length] exists;
+		NativePath[logoFormats.length] paths;
+		foreach (i, format; logoFormats)
+		{
+			paths[i] = NativePath(logoOutputFolder) ~ NativePath.Segment(logo ~ format);
+			exists[i] = existsFile(paths[i]);
+		}
+
+		auto settings = new HTTPFileServerSettings();
+		settings.maxAge = 365.days;
+
+		if (exists[0])
+			sendFile(req, res, paths[0], settings);
```

Comment:
So we just did a lot of file checks, but send the first one anyhow?

Reply:
uh relict from other PR where we had more than 1 file check, we actually just do one file check at the above too, maybe support for jpg or other formats will be added eventually and this is the place to put it